pkg/api/validation/validation.go

```diff
@@ -737,13 +755,14 @@ func ValidateMinionUpdate(oldMinion *api.Node, minion *api.Node) errs.Validation
 // Validate compute resource typename.
 // Refer to docs/resources.md for more details.
 func validateResourceName(str string) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
 	if !util.IsQualifiedName(str) {
-		return errs.ValidationErrorList{fmt.Errorf("invalid compute resource typename format %q", str)}
+		return append(allErrs, errs.NewFieldInvalid("resource typename", str, QualifiedNameErrorMsg))
```

Comment:
Done-ish. I'm not entirely sure what 'fields' to put in places where this function is called. I tried to guess, but PTAL.